### PR TITLE
Add new handlers, and change display names

### PR DESCRIPTION
**Pull request by PanSzelescik**
Add support for:

- Minechem
- Thermal Expansion
- Mekanism
- Ex Nihilo

Change some names:

- minecraft -> Minecraft
- AdvancedSolarPanel -> Advanced Solar Panels
- AppliedEnergistics2 -> Applied Energistics 2
- BloodMagic -> Blood Magic
- EnderIO -> Ender IO
- ExtraCells -> Extra Cells 2
- Forestry -> Forestry for Minecraft
- ExtraUtilities -> Extra Utilities
- gregtech -> GregTech
- IC2 -> IndustrialCraft 2
- WirelessCraftingTerminal -> Wireless Crafting Terminal
- witchery -> Witchery
- Gregtech 6 -> GregTech 6

Change some icons:

- Applied Energistics' Grindstone recipes uses Grindstone icon

### CODE REVIEW

File: resources/assets/nei/csv/handlers.csv [hunk 1]
```diff
@@ -384,4 +384,39 @@ net.dries007.tfcnei.recipeHandlers.KnappingRecipeHandler,TerraFirmaCraft+,terraf
 net.dries007.tfcnei.recipeHandlers.ClothingRecipeHandler,TerraFirmaCraft+,terrafirmacraftplus:item.Bone Needle Thread,,terrafirmacraftplus,TRUE,,,65,166,5,,,,,,
 net.dries007.tfcnei.recipeHandlers.BarrelRecipeHandler,TerraFirmaCraft+,terrafirmacraftplus:Barrel,,terrafirmacraftplus,TRUE,,,65,166,5,,,,,,
 net.dries007.tfcnei.recipeHandlers.AlloyRecipeHandler,TerraFirmaCraft+,terrafirmacraftplus:Crucible,,terrafirmacraftplus,TRUE,,,65,166,5,,,,,,
-net.dries007.tfcnei.recipeHandlers.HeatRecipeHandler,TerraFirmaCraft+,terrafirmacraftplus:Firepit,,terrafirmacraftplus,TRUE,,,65,166,5,,,,,,
\ No newline at end of file
+net.dries007.tfcnei.recipeHandlers.HeatRecipeHandler,TerraFirmaCraft+,terrafirmacraftplus:Firepit,,terrafirmacraftplus,TRUE,,,65,166,5,,,,,,
+minechem.nei.ChemicalBucketNEIRecipeHandler,Minecraft,Mekanism:BasicBlock:8,,minechem,TRUE,,,65,166,5,,,,,,
```
Comment (mitchej123) [inline resources/assets/nei/csv/handlers.csv:388]: This one seems wrong

File: resources/assets/nei/csv/handlers.csv [hunk 2]
```diff
@@ -33,32 +33,32 @@ buildcraft.compat.nei.RecipeHandlerRefinery,BuildCraft,BuildCraft|Factory:refine
...
-crazypants.enderio.nei.EnchanterRecipeHandler,EnderIO,EnderIO:blockEnchanter,,EnderIO,TRUE,,,65,,5,,,,,,
-crazypants.enderio.nei.SagMillRecipeHandler,EnderIO,EnderIO:blockSagMill,,EnderIO,TRUE,,,68,,5,,,,,,
-crazypants.enderio.nei.SliceAndSpliceRecipeHandler,EnderIO,EnderIO:blockSliceAndSplice,,EnderIO,TRUE,,,65,,5,,,,,,
-crazypants.enderio.nei.SoulBinderRecipeHandler,EnderIO,EnderIO:blockSoulBinder,,EnderIO,TRUE,,,68,,5,,,,,,
-crazypants.enderio.nei.VatRecipeHandler,EnderIO,EnderIO:blockVat,,EnderIO,TRUE,,,65,,5,,,,,,
-extracells.integration.nei.UniversalTerminalRecipe,ExtraCells,extracells:part.base:9,,extracells,TRUE,,,65,,5,,,,,,
-com.rwtema.extrautils.nei.EnderConstructorHandler,ExtraUtilities,ExtraUtilities:endConstructor,,ExtraUtilities,TRUE,,,68,,5,,,,,,
-com.rwtema.extrautils.nei.FMPMicroBlocksHandler,ExtraUtilities,ForgeMicroblock:sawIron,,ExtraUtilities,TRUE,,,65,,5,,,,,,
-com.rwtema.extrautils.nei.InfoHandler,ExtraUtilities,ExtraUtilities:chandelier,,ExtraUtilities,TRUE,,,135,,2,,,,,,"Info on blocks, can vary wildly. EQ was this length, there might be a longer one"
```
Comment (mitchej123) [inline resources/assets/nei/csv/handlers.csv:45]: The quotes shouldn't be removed on the strings, especially the ones that include a comma

File: resources/assets/nei/csv/handlers.csv [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-fox.spiteful.avaritia.compat.nei.ExtremeShapelessRecipeHandler,Avaritia,Avaritia:Dire_Crafting,,Avaritia,TRUE,,,208,256,1,,,,,,"If you can cut off/place over the jaggedness at the bottom, you could barely get a second one"
-WayofTime.alchemicalWizardry.client.nei.NEIAlchemyRecipeHandler,BloodMagic,AWWayofTime:blockWritingTable,,AWWayofTime,TRUE,,,68,,5,,,,,,
-WayofTime.alchemicalWizardry.client.nei.NEIAltarRecipeHandler,BloodMagic,AWWayofTime:Altar,,AWWayofTime,TRUE,,,65,,5,,,,,,Handler name needs shifting left
-WayofTime.alchemicalWizardry.client.nei.NEIBindingRitualHandler,BloodMagic,AWWayofTime:sacrificialKnife,,AWWayofTime,TRUE,,,50,,6,,,,,,
-WayofTime.alchemicalWizardry.client.nei.NEIBloodOrbShapedHandler,BloodMagic,AWWayofTime:AlchemicalWizardrybloodRune:3,,AWWayofTime,TRUE,,,65,,5,,,,,,
-WayofTime.alchemicalWizardry.client.nei.NEIBloodOrbShapelessHandler,BloodMagic,AWWayofTime:AlchemicalWizardrybloodRune:4,,AWWayofTime,TRUE,,,65,,5,,,,,,
+fox.spiteful.avaritia.compat.nei.ExtremeShapelessRecipeHandler,Avaritia,Avaritia:Dire_Crafting,,Avaritia,TRUE,,,208,256,1,,,,,,If you can cut off/place over the jaggedness at the bottom, you could barely get a second on
```
Comment (mitchej123) [inline resources/assets/nei/csv/handlers.csv:19]: don't remove the quotes

File: resources/assets/nei/csv/handlers.csv [hunk 4]
```diff
@@ -234,112 +234,112 @@ tconstruct.plugins.nei.RecipeHandlerCastingBasin,Tinker's Construct,TConstruct:S
...
-alkalus.main.nei.NEI_Handler_Oven,witchery,witchery:witchesovenidle,,WitcheryExtras,TRUE,,,,,,,,,,,Not working at all now
-alkalus.main.nei.NEI_Handler_SpinningWheel,witchery,witchery:spinningwheel,,WitcheryExtras,TRUE,,,65,,5,,,,,,
-com.emoniph.witchery.integration.NEICauldronRecipeHandler,witchery,witchery:cauldron,,witchery,TRUE,,,65,,5,,,,,,Replaced by WE presumably
-com.emoniph.witchery.integration.NEIDistilleryRecipeHandler,witchery,witchery:distilleryidle,,witchery,TRUE,,,140,,2,,,,,,Replaced by WE presumably
-com.emoniph.witchery.integration.NEIKettleRecipeHandler,witchery,witchery:kettle,,witchery,TRUE,,,65,,5,,,,,,Replaced by WE presumably
-com.emoniph.witchery.integration.NEISpinningWheelRecipeHandler,witchery,witchery:spinningwheel,,witchery,TRUE,,,,,,,,,,,Replaced by WE presumably
-com.emoniph.witchery.integration.NEIWitchesOvenRecipeHandler,witchery,witchery:witchesovenidle,,witchery,TRUE,,,65,,5,,,,,,Replaced by WE presumably
+tconstruct.plugins.nei.RecipeHandlerToolMaterials,Tinker's Construct,TConstruct:ToolStationBlock,,TConstruct,TRUE,,,110,,3,,,,,,Longer if more modifiers, I went through the whole list
```
Comment (mitchej123) [inline resources/assets/nei/csv/handlers.csv:237]: quotes

**Review by mitchej123 — CHANGES_REQUESTED** [review]
Thanks for the additions.  A few small issues to cleanup before I can merge it.

File: resources/assets/nei/csv/handlers.csv (hunk 1 above)
Comment (PanSzelescik) [reply resources/assets/nei/csv/handlers.csv]: Oh, right

File: resources/assets/nei/csv/handlers.csv (hunk 2 above)
Comment (PanSzelescik) [reply resources/assets/nei/csv/handlers.csv:45]: Yes, fixed

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**